src/tests/zf_apps/zfsink.c

```diff
@@ -233,8 +247,51 @@ void print_attrs(struct zf_attr* attr)
 }
 
 
+static void monitor(struct resources* res)
+{
+  uint64_t now_bytes, prev_bytes;
+  struct timeval start, end;
+  uint64_t prev_pkts, now_pkts;
+  int ms, pkt_rate, mbps;
+
+  printf("#%9s %16s %16s", "pkt-rate", "bandwidth(Mbps)", "total-pkts\n");
+
+  prev_pkts = res->n_rx_pkts;
+  prev_bytes = res->n_rx_bytes;
+  gettimeofday(&start, NULL);
+
+  while( 1 ) {
+    sleep(1);
+    now_pkts = res->n_rx_pkts;
+    now_bytes = res->n_rx_bytes;
```

Comment:
Also, because it's shared data, res should at least be volatile. If you want to be completely correct, you should use [atomic builtins](https://gcc.gnu.org/onlinedocs/gcc/_005f_005fatomic-Builtins.html) with relaxed ordering.